pgx-utils/src/sql_entity_graph/pg_trigger/mod.rs

```diff
@@ -11,46 +11,51 @@ use syn::{ItemFn, Token};
 #[derive(Debug, Clone)]
 pub struct PgTrigger {
     func: syn::ItemFn,
-    attributes: syn::punctuated::Punctuated<PgTriggerAttribute, Token![,]>,
+    to_sql_config: ToSqlConfig,
 }
 
 impl PgTrigger {
     pub fn new(
         func: ItemFn,
         attributes: syn::punctuated::Punctuated<PgTriggerAttribute, Token![,]>,
     ) -> Result<Self, syn::Error> {
-        Ok(Self { func, attributes })
-    }
 
-    pub fn to_sql_config(&self) -> Result<Option<ToSqlConfig>, syn::Error> {
-        let mut found = None;
-        for attribute in self.attributes.iter() {
-            match attribute {
-                &PgTriggerAttribute::Sql(ref to_sql_config) if found.is_none() => {
-                    found = Some(to_sql_config.clone())
-                }
-                &PgTriggerAttribute::Sql(_) if found.is_some() => {
-                    return Err(syn::Error::new(
-                        Span::call_site(),
-                        "Multiple `sql` arguments found, it must be unique",
-                    ))
+        let to_sql_config = {
+            let mut found = None;
+            for attribute in attributes.iter() {
+                match attribute {
+                    &PgTriggerAttribute::Sql(ref to_sql_config) if found.is_none() => {
+                        found = Some(to_sql_config.clone())
+                    }
+                    &PgTriggerAttribute::Sql(_) if found.is_some() => {
+                        return Err(syn::Error::new(
+                            Span::call_site(),
+                            "Multiple `sql` arguments found, it must be unique",
+                        ))
+                    }
+                    _ => (),
                 }
-                _ => (),
             }
-        }
 
-        if let Some(ref mut found) = found {
-            if let Some(ref mut content) = found.content {
-                let value = content.value();
-                let updated_value = value.replace(
-                    "@FUNCTION_NAME@",
-                    &*(self.func.sig.ident.to_string() + "_wrapper"),
-                ) + "\n";
-                *content = syn::LitStr::new(&updated_value, Span::call_site());
+            if let Some(ref mut found) = found {
+                if let Some(ref mut content) = found.content {
+                    let value = content.value();
+                    let updated_value = value.replace(
+                        "@FUNCTION_NAME@",
+                        &*(func.sig.ident.to_string() + "_wrapper"),
+                    ) + "\n";
+                    *content = syn::LitStr::new(&updated_value, Span::call_site());
+                }
             }
+
+            found.unwrap_or_default()
+        };
```

Comment:
I found this control flow confusing to follow. Maybe the final unwrap pushed it over the edge for my brain. Would something like this be okay?

```diff
diff --git a/pgx-utils/src/sql_entity_graph/pg_trigger/mod.rs b/pgx-utils/src/sql_entity_graph/pg_trigger/mod.rs
index d4b9e0b..c9143e9 100644
--- a/pgx-utils/src/sql_entity_graph/pg_trigger/mod.rs
+++ b/pgx-utils/src/sql_entity_graph/pg_trigger/mod.rs
@@ -19,37 +19,22 @@ impl PgTrigger {
         func: ItemFn,
         attributes: syn::punctuated::Punctuated<PgTriggerAttribute, Token![,]>,
     ) -> Result<Self, syn::Error> {
-
-        let to_sql_config = {
-            let mut found = None;
-            for attribute in attributes.iter() {
-                match attribute {
-                    &PgTriggerAttribute::Sql(ref to_sql_config) if found.is_none() => {
-                        found = Some(to_sql_config.clone())
-                    }
-                    &PgTriggerAttribute::Sql(_) if found.is_some() => {
-                        return Err(syn::Error::new(
-                            Span::call_site(),
-                            "Multiple `sql` arguments found, it must be unique",
-                        ))
-                    }
-                    _ => (),
-                }
-            }
-
-            if let Some(ref mut found) = found {
-                if let Some(ref mut content) = found.content {
+        if attributes.len() > 1 {
+            return Err(syn::Error::new(
+                Span::call_site(),
+                "Multiple `sql` arguments found, it must be unique",
+            ))
+        };
+        let to_sql_config = attributes.first().cloned().map(|PgTriggerAttribute::Sql(mut config)| {
+                if let Some(ref mut content) = config.content {
                     let value = content.value();
-                    let updated_value = value.replace(
-                        "@FUNCTION_NAME@",
-                        &*(func.sig.ident.to_string() + "_wrapper"),
-                    ) + "\n";
+                    let updated_value = value
+                        .replace("@FUNCTION_NAME@", &*(func.sig.ident.to_string() + "_wrapper"))
+                        + "\n";
                     *content = syn::LitStr::new(&updated_value, Span::call_site());
-                }
-            }
-
-            found.unwrap_or_default()
-        };
+                };
+                config
+            }).unwrap_or_default();
 
         if !to_sql_config.overrides_default() {
             crate::ident_is_acceptable_to_postgres(&func.sig.ident)?;
```